Fix trailing whitespace

tacshi opened the pull request:
Closes [#ISSUE](https://github.com/zed-industries/zed/issues/45407)

Release Notes:

- Fixed `remove_trailing_whitespace_on_save` configuration

Recording (outdated, see new commit below):


https://github.com/user-attachments/assets/dde66213-fc68-4f1f-81ba-d1905f00be07

Commit:
Add `remove_whitespace_on_empty_lines` setting to control whitespace removal on empty lines during save

tacshi commented:
The latest commit respects the proposition of @yeskunall 

Recording:

https://github.com/user-attachments/assets/bc19727e-8505-475a-b3e2-377422b15f3e

maxdeviant reviewed (changes requested):
I don’t think the analysis in that issue that this is a bug is correct.

Removing whitespace from every single line is how I would expect this feature to work.

If we want to support this alternate behavior, it would need to be behind a separate setting.

tacshi commented:
> I don’t think the analysis in that issue that this is a bug is correct.
> 
> Removing whitespace from every single line is how I would expect this feature to work.
> 
> If we want to support this alternate behavior, it would need to be behind a separate setting.

It seems there's some disagreement regarding certain functionalities.
How was the first version of the screen recording? I'd be happy to revert to that version and submit it.

tacshi commented:
**Hard** reset to the first commit.
@maxdeviant

Commit:
Fix `ensure_final_newline` regression and VS Code import mapping for `remove_whitespace_on_empty_lines`

- Scope `remove_whitespace_on_empty_lines` in `ensure_final_newline` to only
  take effect when `remove_trailing_whitespace_on_save` is also enabled,
  matching the setting's documented behavior
- Fix inverted VS Code `trimAutoWhitespace` mapping (true should map to true)
- Update `ensure_final_newline` doc comment to reflect conditional behavior

Commit:
update a comment

Commit:
Merge branch 'main' into fix/whitespace

Commit:
Merge branch 'main' into fix/whitespace

MrSubidubi commented:
<img width="979" height="466" alt="image" src="https://github.com/user-attachments/assets/a379f75c-017e-4a24-a7fd-8febd019da3f" />

Will close given the current state of things. Feel free to resubmit after the comments have been adressed and everything is green again. Thanks!
